### PR TITLE
Script to record videos yourself

### PR DESCRIPTION
**Pull request by manikd31**
## AV-3714: Script to record videos yourself
- Ticket: [AV-3714](https://20bn.atlassian.net/browse/AV-3714)

### Developer Notes:
This PR updates the `video_capture.py` script that can be used to record videos ourselves. 

**Acceptance criteria**:
- [x] Ensure that pre-existing videos are not overwritten
- [x] Refactor code for simplification
- [x] FPS issue
- [x] Escape character to stop process _(optional)_
- [x] Escape character to skip current video without saving _(optional)_

### CODE REVIEW

File: scripts/tools/video_capture.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+    # Calculate time taken (in seconds) to read `number_frames`
+    start_time = time.time()
+    for n in range(number_frames):
+        _, _ = video.read()
+    end_time = time.time()
+
+    calculated_fps = int(number_frames/(end_time - start_time))
+
+    return 16. if calculated_fps <= 16 else calculated_fps
```
Comment (guillaumebrg) [inline scripts/tools/video_capture.py:48]: Is a float acceptable here?

**Review by guillaumebrg — COMMENTED** [review]
Tested it, and it worked. I made a bunch of minor comments. 

Here are a few suggestions:

- instead of using the `_get_fps` trick to determine the camera fps, `_capture_video` could have two loops: a `while` loop to get and store all the frames followed by a `for` loop to encode all the frames and we would set the fps of the VideoWriter using `round(video_duration / len(frames))`

- We should use the "escape" key rather than "q" to exit because that's what we use in the live demos. Rather than skipping the current video, I think pressing "escape" should exit the whole thing... and the current video shouldn't be saved

File: scripts/tools/video_capture.py (hunk 1 above)
Comment (manikd31) [reply scripts/tools/video_capture.py:48]: I didn't mean to put that as a float, but it didn't cause any issue. To be on a safer side, I'll change that to an integer.

File: scripts/tools/video_capture.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    cv2.destroyAllWindows()
+
+
+if __name__ == "__main__":
+    # Parse arguments
+    args = docopt(__doc__)
+    duration = float(args['--duration'])
+    pre_recording_duration = float(args['--pre_recording_duration'])
+    number_videos = int(args['--number_videos'])
+    camera_id = args['--camera_id'] or 0
```
[the remark this answers is not in the document]
Comment (manikd31) [reply scripts/tools/video_capture.py:103]: I figured that should be done as well.

File: scripts/tools/video_capture.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+               [--pre_recording_duration=PRE_RECORDING_DURATION]
+               [--number_videos=NUMBER_VIDEOS]
+               [--camera_id=CAMERA_ID]
+               [--path_out=PATH_OUT]
+               [--file_name=FILE_NAME]
+  video_capture.py (-h | --help)
+
+Options:
+  --duration=DURATION                                   Recording time in seconds
+  --pre_recording_duration=PRE_RECORDING_DURATION       Duration for the pre-recording period
```
[the remark this answers is not in the document]
Comment (manikd31) [reply scripts/tools/video_capture.py:15]: Do you mean a default time for both `pre-recording` and `recording` durations?

File: scripts/tools/video_capture.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    # Avoid overwriting pre-existing files
+    while file in os.listdir(path_out):
+        i += 1
+        file = f"{filename}_{str(i)}.mp4"
+
+    # Show timer window before recording
+    _capture_video(video_duration=pre_recording_duration)
+
+    # Show timer window for actual recording
+    _capture_video(video_duration=duration, record=True)
```
[the remark this answers is not in the document]
Comment (manikd31) [reply scripts/tools/video_capture.py:123]: This was this way before I started editing it. I felt this should be inside the main as well and not global.

File: scripts/tools/video_capture.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (guillaumebrg) [reply scripts/tools/video_capture.py:15]: `pre-recording`

**Comment by manikd31** [comment]
Key-strokes handling:
- Pressing `Q` will skip the current video prompt and move to the next (skipping during pre-recording phase will jump to the recording phase and not skip the entire process for the current video)
- _(need to implement)_ Pressing `Esc` should skip the whole process without saving any files currently being written (excpet those already written)

**Review by guillaumebrg — APPROVED** [review]
:fire:

**Pull request merged**